### PR TITLE
allow custom auth endpoint

### LINKED ISSUE #1285: Troubleshooting oauth client
**- Do you want to request a *feature* or report a *bug*?**
Might be a bug, might be a feature 

**- What is the current behavior?**

While implementing an OAuth auth backend for Github, I'm seeing what might be an issue either with Redux state or inter-window communication.

My code:
https://github.com/marksteele/netlify-serverless-oauth2-backend

The HTML bits are copied from the other implementations (go/python/nodejs), and the HTML output appears to succeed, however the auth window does not close and the app state doesn't update to reflect the successful auth. Might be a bug in my code...

**- If the current behavior is a bug, please provide the steps to reproduce.**

**- What is the expected behavior?**
Auth window closes, react app updates with token from auth success.

**- Please mention your versions where applicable.**
netflify 1.5.0

https://www.control-alt-del.org/admin/

**- Please link or paste your `config.yml` below if applicable.**
https://www.control-alt-del.org/admin/config.yml

### PR DESCRIPTION
**Pull request by erquhart**
**- Summary**

Currently the path "/auth" is enforced for authentication, even with a custom `base_url`. Developers should be able to override the default endpoint.

Fixes #1285.

**- Description for the changelog**

allow custom auth endpoint

### CODE REVIEW

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit 55ff84f94e5f979c84e0d3cd93c72195c6b0be4f

https://deploy-preview-1294--cms-demo.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 55ff84f94e5f979c84e0d3cd93c72195c6b0be4f

https://deploy-preview-1294--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit cfb290779144e0ef12391370586c5aa96ef94aac

https://deploy-preview-1294--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit cfb290779144e0ef12391370586c5aa96ef94aac

https://deploy-preview-1294--cms-demo.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 8cc6bb3534aa66ef852e5cbfad4009352bf94469

https://deploy-preview-1294--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit 8cc6bb3534aa66ef852e5cbfad4009352bf94469

https://deploy-preview-1294--cms-demo.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit fb082fc517d17f1cc0dc23173d7ec59b844e214f

https://deploy-preview-1294--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit fb082fc517d17f1cc0dc23173d7ec59b844e214f

https://deploy-preview-1294--cms-demo.netlify.com

**Comment by erquhart** [comment]
@marksteele please test and confirm this works for your use case.

**Comment by tech4him1** [comment]
@erquhart I'm good with this. I do wonder, however, if it would just be better to have a new setting like `auth_url` that would replace the `base_url` and `authEndpoint` at once. `base_url` is already really unclear in my opinion.

**Comment by erquhart** [comment]
I considered that, but base_url is used in a few places. We'd need to make sure it's okay to get rid of.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by tech4him1** [comment]
Hmm, that may be necessary, but I only see it used with `netlify-auth.js` (and the components that forward the value to it).

**Comment by erquhart** [comment]
I believe we're using it to test against the `origin` property of the [`MessageEvent`](https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent) from the popup, so we need the stripped origin. We could do this internally, though.

I'm inclined to go with what's in this PR since it's backward compatible, and maybe provide the more ideal solution you're describing in a 2.0 PR. What do you think?

**Comment by tech4him1** [comment]
2.0 is fine, go for it.

File: src/components/App/App.js [hunk 1]
```diff
@@ -84,6 +84,7 @@ class App extends React.Component {
             isFetching: auth && auth.get('isFetching'),
             siteId: this.props.config.getIn(["backend", "site_domain"]),
             base_url: this.props.config.getIn(["backend", "base_url"], null),
+            authEndpoint: this.props.config.getIn(["backend", "auth_endpoint"]),
```
Comment (tech4him1) [inline src/components/App/App.js]: I wonder if we should make backends get the value themselves, instead of passing them all in here. Maybe a 2.0 thing?

Comment (erquhart) [reply src/components/App/App.js]: 2.0, but yeah that definitely makes sense. @Benaiah what do you think?

**Pull request merged**